### ISSUE TITLE
Bedevere should label "awaiting merge" after a core review

### ISSUE THREAD

**Issue by hugovk** [python/bedevere#584]
# Steps to reproduce

1. PR has "awaiting review" label
2. A core dev approves it

# Actual result

Bedevere removes "awaiting review" and adds "awaiting core review"

(This is what should happen when a non-core dev approves.)

# Expected result

Bedevere removes "awaiting review" and adds "awaiting merge"

For [example](https://github.com/python/cpython/pull/109753):

![image](https://github.com/python/bedevere/assets/1324225/d3e0fc14-0ceb-491c-b0b2-0528c249039a)

And:

![image](https://github.com/python/bedevere/assets/1324225/c06abc5b-f871-4dec-ae60-8cfc28dcbe86)

# See also

State diagram at https://github.com/python/bedevere#pr-state-machine

**Comment by Mariatta** [comment]
I'll take a look tomorrow.

**Comment by pradyunsg** [comment]
@Mariatta and I investigated this yesterday, and this has been resolved.

For posterity: The bot was recently migrated from a GitHub (bot) account to a GitHub App. During this migration, insufficient permissions were provided to the bedevere GitHub App -- it did not have the permission to query the membership to teams within the `python` organisation. This meant that the bot could not determine who is a core team member, and would label PRs as if everyone reviewing PRs is not a core developer.

Adding the relevant permission resolved this issue.
